### ISSUE TITLE
@XmlIgnoreWhitespace is not working for Xml Value 

### ISSUE THREAD

**Issue by Dhanush1801** [pdvrieze/xmlutil#120]
The deserialising Xml
```
<A>
        <B><text xml:space="preserve">Cc. </text></B>
	<B><text xml:space="preserve">        </text></B>
	<B><text xml:space="preserve">   </text></B>
	<B><text xml:space="preserve">h ello </text></B>
</A>

```
corresponding data class
```
@Serializable
data class A(
    @SerialName("B")
    val b: List<B>
)
@Serializable
data class B(
    @XmlSerialName("text", "", "")
    val text: Text
)
@Serializable
data class Text(
    @XmlValue(true)
    @XmlIgnoreWhitespace(false)
    var value: String? = null,
    @XmlElement(false)
    @XmlSerialName("space", "http://www.w3.org/XML/1998/namespace", "")
    val space: String? = null
)

```
**while deserialising Xml whitespace is ignored for XmlValue**

Actual Output:
`A(b=[B(text=Text(value=Cc. , space=preserve)), B(text=Text(value=null, space=preserve)), B(text=Text(value=null, space=preserve)), B(text=Text(value=h ello , space=preserve))])`

Expected output:
`A(b=[B(text=Text(value=Cc. , space=preserve)), B(text=Text(value=           , space=preserve)), B(text=Text(value=    , space=preserve)), B(text=Text(value=h ello ,  #space=preserve))])`

**Is there is a way to get ignored whitespace(without text) while deserialising ? **

Note: if the XmlValue field type is List of String type whitespaces captured from the xml while deserialising

**Comment by pdvrieze** [comment]
I've managed to fix this. Note that preserving spaces is the default behaviour but there is an error in handling it if the text content starts with whitespace. In general the preservation of whitespace for strings was incomplete. This has now been resolved. In addition the decoding (and encoding) will now detect the preserve property automatically.

Please note that emitting the xml:space property by yourself as serialization attribute isn't supported (it may break), the decoder will now process the value itself and not pass it for deserialization. The encoder will emit it if needed.

**Comment by Dhanush1801** [comment]
Thanks!!
i have tested the recent snapshot(0.84.4-SNAPSHOT) build but it has some  issues
xml=`<A>
	<B>
		<text xml:space="preserve">Cc. </text>
	</B>
	<B>
		<text xml:space="preserve">f</text>
	</B>
	<B>
		<text xml:space="preserve">     </text>
	</B>
	<B>
		<text xml:space="default">  hello</text>
	</B>
</A>`

data class:
`@Serializable
data class A(
    @SerialName("B")
    val b: List<B>
)
@Serializable
data class B(
    @XmlElement(true)
    @XmlSerialName("text", "", "")
    val text: Text
){

}
@Serializable
data class Text(
    @XmlValue(true)
    @XmlIgnoreWhitespace(false)
    var value:String,
)`

**@XmlIgnoreWhitespace(true) preserves whitespaces and while encoding the output xml contains xml:space has attribute if xml has whitespace**

 @XmlIgnoreWhitespace(true) Output:
```
decoded Data Class Obj:A(b=[B(text=Text(value=Cc. )), B(text=Text(value=f)), B(text=Text(value=     )), B(text=Text(value=  hello))])
generated XMl:<A><B><text xml:space="preserve">Cc. </text></B><B><text>f</text></B><B><text xml:space="preserve">     </text></B><B><text xml:space="preserve">  hello</text></B></A>
```

**@XmlIgnoreWhitespace(false) does not emits the whitespaces and encoding the generated xml does not contain xml:space**

@XmlIgnoreWhitespace(false) Ouput:

```
decoded Data Class Obj:A(b=[B(text=Text(value=Cc. )), B(text=Text(value=f)), B(text=Text(value=     )), B(text=Text(value=  hello))])
generated XMl:<A><B><text>Cc. </text></B><B><text>f</text></B><B><text>     </text></B><B><text>  hello</text></B></A>
```

**Comment by pdvrieze** [comment]
I've added this test with your data types. It works for me. Perhaps you don't have the most recent snapshot (I've checked sonatype, the latest version - the one you need is has the date 20230210 as part of the artefact names)

**Comment by pdvrieze** [comment]
It should be fixed now in [v0.85.0](https://github.com/pdvrieze/xmlutil/releases/tag/v0.85.0).
